src/gui/modalMenu.cpp

```diff
@@ -230,89 +288,50 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 	}
 
 	if (event.EventType == EET_TOUCH_INPUT_EVENT) {
-		SEvent translated;
-		memset(&translated, 0, sizeof(SEvent));
-		translated.EventType = EET_MOUSE_INPUT_EVENT;
-		gui::IGUIElement *root = Environment->getRootGUIElement();
-
-		if (!root) {
-			errorstream << "GUIModalMenu::preprocessEvent"
-				    << " unable to get root element" << std::endl;
-			return false;
-		}
-		gui::IGUIElement *hovered =
-				root->getElementFromPoint(core::position2d<s32>(
-						event.TouchInput.X, event.TouchInput.Y));
-
-		translated.MouseInput.X = event.TouchInput.X;
-		translated.MouseInput.Y = event.TouchInput.Y;
-		translated.MouseInput.Control = false;
+		irr_ptr<GUIModalMenu> holder;
+		holder.grab(this); // keep this alive until return (it might be dropped downstream [?])
 
-		if (event.TouchInput.touchedCount == 1) {
-			switch (event.TouchInput.Event) {
-			case ETIE_PRESSED_DOWN:
+		switch ((int)event.TouchInput.touchedCount) {
```

Comment:
why the int cast here?

Reply:
because that is a float in some IrrLicht version, for whatever weird reason